src/include/thread_inline.h

```diff
@@ -0,0 +1,99 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
```

Comment:
These are very rarely used functions. They don't need to be in a new inlined header file. I'd also suggest you look at `support/thread_group.c` to see if there is any overlap that can be gained or code reuse (i.e. a "thread group of 1"). You're using the structure of threads in the thread groups so there may be more overlap. At least these "individual thread" functions should be in that source file instead.

Reply:
Yep the duplicated thread logic was my main concern with the change so far. I'll see if I can reuse `thread_group` for this. This ticket is part of a mini-spike so I'll time-box it, and if I can't find a sufficiently clean implementation end of day I'll close the PR and we can revisit it in the project proposed in WRITING-9312.